PWGCF/Flow/Tasks/flowPtEfficiency.cxx

```diff
@@ -71,13 +95,46 @@ struct flowPtEfficiency {
 
     registry.add("mcEventCounter", "Monte Carlo Truth EventCounter", kTH1F, {axisCounter});
     registry.add("hPtMCGen", "Monte Carlo Truth", {HistType::kTH1D, {axisPt}});
+
+    if (cfgTrkSelRun3ITSMatch) {
+      myTrackSel = getGlobalTrackSelectionRun3ITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSall7Layers, TrackSelection::GlobalTrackRun3DCAxyCut::Default);
+    } else {
+      myTrackSel = getGlobalTrackSelectionRun3ITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSibAny, TrackSelection::GlobalTrackRun3DCAxyCut::Default);
+    }
+    if (cfgCutDCAxyppPass3Enabled) {
+      myTrackSel.SetMaxDcaXYPtDep([](float pt) { return 0.004f + 0.013f / pt; });
+    } else {
+      myTrackSel.SetMaxDcaXY(cfgCutDCAxy);
+    }
+    myTrackSel.SetMinNClustersTPC(cfgCutTPCclu);
+    myTrackSel.SetMinNCrossedRowsTPC(cfgCutTPCcrossedrows);
+    if (!cfgCutDCAzPtDepEnabled)
+      myTrackSel.SetMaxDcaZ(cfgCutDCAz);
   }
 
-  void processReco(o2::aod::Collision const&, myTracks const& tracks, aod::McParticles const&)
+  template <typename TTrack>
+  bool trackSelected(TTrack track)
+  {
+    if (cfgCutDCAzPtDepEnabled && (track.dcaZ() > (0.004f + 0.013f / track.pt())))
+      return false;
+    return myTrackSel.IsSelected(track);
+  }
+
+  void processReco(myCollisions::iterator const& collision, aod::BCsWithTimestamps const&, myTracks const& tracks, aod::McParticles const&)
   {
     registry.fill(HIST("eventCounter"), 0.5);
+    if (!collision.sel8())
+      return;
+    if (tracks.size() < 1)
+      return;
+    if (cfgSelRunNumberEnabled) {
+      auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+      int RunNumber = bc.runNumber();
+      if (!std::count(cfgRunNumberList.value.begin(), cfgRunNumberList.value.end(), RunNumber))
```

Comment:
Doing this on a per `bc` level may incur in a high CPU usage depending on the number of runs to consider
If the run number has already been checked and the coming `bc` brings the same run number there is not point for checking it again
Please, consider it for the next iteration